compiler/rustc_feature/src/unstable.rs

```diff
@@ -425,8 +425,8 @@ declare_features! (
     (unstable, deprecated_suggestion, "1.61.0", Some(94785)),
     /// Allows deref patterns.
     (incomplete, deref_patterns, "1.79.0", Some(87121)),
-    /// Allows deriving `SmartPointer` traits
-    (unstable, derive_smart_pointer, "1.79.0", Some(123430)),
+    /// Allows deriving traits as per `CoercePointee` specification
+    (unstable, derive_coerce_pointee, "1.79.0", Some(123430)),
```

Comment:
Why is this a lang feature at all? The feature gating should be a library feature, since all this gates is a macro, and no behavior in the compiler.

I believe that you can remove this.